src/services/gtm.ts

```diff
@@ -12,3 +15,18 @@ export const initGoogleTagManager = () => {
         TagManager.initialize({ gtmId: googleTagManagerSettings.id });
     }
 };
+
+export const fireGtmEvent = (event: EVENTS, data: Schema | undefined = {}) => {
+    if (
+        isProduction &&
+        googleTagManagerSettings.enabled &&
+        googleTagManagerSettings.id
+    ) {
```

Comment:
Duplicating check for ease of approach. Needs to later be shared with the function above